### PR TITLE
Do not use keystone CLI to fetch service tenant ID

### PR DESCRIPTION
**Pull request by vuntz**
On the initial run, we might not have the keystone client installed
(that's the case if the node has not been used for anything before this
recipe is run).

### CODE REVIEW

**Comment by vuntz** [comment]
Note that this is really ugly. I was wondering about doing something like

```ruby
package python-keystoneclient
  action :nothing
end.run_action(:install)
```

But I don't like it much either :/

**Comment by rhafer** [comment]
> But I don't like it much either :/

Why not? I think it's a lot better than the approach you implemented. ;)

**Comment by vuntz** [comment]
@rhafer okay, updated for the other approach

File: chef/cookbooks/neutron/recipes/common_config.rb [hunk 1]
```diff
@@ -126,6 +126,11 @@
     action :nothing
   end.run_action(:wakeup)
 
+  # Ensure the keystone call below will not fail because of the missing CLI
+  package python-keystoneclient
```
Comment (rhafer) [inline chef/cookbooks/neutron/recipes/common_config.rb]: Travis fails. ```do``` is missing here.

**Comment by vuntz** [comment]
Reworked this code path again, after I found out a reboot could break things and lead the node to go into problem state.

File: chef/cookbooks/neutron/recipes/common_config.rb (hunk 1 above)
Comment (vuntz) [inline chef/cookbooks/neutron/recipes/common_config.rb]: Oops, sorry, fixed.

**Comment by dirkmueller** [comment]
IMHO thats super ugly.. how about recording the admin tenant id in keystone barclamp and just using the value from chef there? its not going to change anyway, the whole querying that thing dynamically was a mistake to begin with..

**Comment by vuntz** [comment]
@dirkmueller: see https://github.com/crowbar/barclamp-keystone/issues/264 :-)

**Comment by dirkmueller** [comment]
@vuntz can you please rework this after crowbar/barclamp-keystone#264 got merged?

**Comment by vuntz** [comment]
@dirkmueller done, but not tested yet.

**Comment by rhafer** [comment]
+1, survived a quick test

**Pull request merged**